### PR TITLE
V17  save matches array

### PR DESCRIPTION
**Pull request by eliasYazdani**

### CODE REVIEW

File: backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/Round.java
Comment (AlexanderLeonidasGuenzel) [inline backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/Round.java:1]: too many empty lines between curly brackets

File: backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentController.java [hunk 1]
```diff
@@ -24,19 +24,19 @@ Tournament addNewTournament(@Valid @RequestBody TournamentWithoutID tournamentWi
         return tournamentService.addNewTournament(tournamentWithoutID);
     }
 
-    @GetMapping("/{id}")
-    public Tournament getDetails(@PathVariable String id) {
-        return tournamentService.getDetailsById(id);
```
Comment (AlexanderLeonidasGuenzel) [inline backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentController.java:29]: shifting

File: backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentService.java [hunk 2]
```diff
@@ -22,24 +22,33 @@ public List<Tournament> getAllTournamentsForAdmin(String admin) {
     }
 
     public Tournament addNewTournament(TournamentWithoutID tournamentWithoutID) {
-        List<Match> newMatches = new ArrayList<>();
```
Comment (AlexanderLeonidasGuenzel) [inline backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentService.java:25]: I wouldn't take a keyword like _new_ for naming it. An appropriate name is just _rounds_ for this list.

File: backend/src/test/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentIntegrationTest.java [hunk 3]
```diff
@@ -105,31 +111,32 @@ void expectSearchedTournament_whenGetRequestWithIdTournament() throws Exception
...
-                                                   {"id":"M3","player1":"A","score1":2,"player2": "C","score2": 0}],
-                                        "champion": "A"
+                                        "rounds": [
+                                         {"id":"R1","matches":[{"id":"M1","player1":"E","score1":3,"player2": "F","score2": 2},
+                                                               {"id":"M2","player1":"G","score1":1,"player2": "G","score2": 3}]},
+                                         {"id":"R2","matches":[{"id":"M3","player1":"E","score1":2,"player2": "G","score2": 3}]}],
+                                        "champion": "G"
                                     }
                                 """)
-                                .with(csrf())
```
Comment (AlexanderLeonidasGuenzel) [inline backend/src/test/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentIntegrationTest.java:132]: shifting

File: frontend/src/TournamentComponents/Match.tsx [hunk 4]
```diff
@@ -21,14 +22,14 @@ export default function Match(propsMatch: PropsMatch) {
...
                 players={propsMatch.players}
-                onScoreChange1={propsMatch.onScoreChange1} // Pass the matchIndex along with player and score
+                onScoreChange1={propsMatch.onScoreChange1}
                 onPlayerChange1={propsMatch.onPlayerChange1}
                 score1={propsMatch.score1}
                 player1={propsMatch.player1}
             />
             <PlayerWithScore2
                 players={propsMatch.players}
-                onScoreChange2={propsMatch.onScoreChange2} // Pass the matchIndex along with player and score
```
Comment (AlexanderLeonidasGuenzel) [inline frontend/src/TournamentComponents/Match.tsx:31]: comments

File: frontend/src/Models/RoundWithoutId.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+import {MatchWithoutId} from "./MatchWithoutId.ts";
+
+export type RoundWithoutId = {
+    matchesWithoutId: MatchWithoutId[],
+}
```
Comment (AlexanderLeonidasGuenzel) [inline frontend/src/Models/RoundWithoutId.ts]: missing empty line at EOF

**Review by AlexanderLeonidasGuenzel — APPROVED** [review]
clean code. check shifting, spaces, empty lines, comments, ...

File: backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/Round.java
Comment (eliasYazdani) [reply backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/Round.java:1]: 👍danke

File: backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentService.java (hunk 2 above)
Comment (eliasYazdani) [reply backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentService.java:25]: I am creating new Tournament with new rounds and metches. Thanks for your suggestion.

**Pull request merged**